app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
+function findMatchingClosingBracketIndex(source, fromIndex) {
+  let result = -1
+  let currentDepth = 1
+  for (let i = fromIndex; i < source.length; i++) {
+    const char = source[i]
+    if (char === "{") {
+      currentDepth++
+    } else if (char === "}") {
+      currentDepth--
+      if (currentDepth <= 0) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
+  const conditionalStartRegex = /@if *\( *((?:.|\n)+?) *(==|!=) *((?:.|\n)+?)\) *[ \n]*\{/g
+  const conditionalElseStartRegex = / *@else *\{/
+
+  let match
+  while ((match = conditionalStartRegex.exec(source)) != null) {
+    const [matchedConditionalStartText, left, operation, right] = match
+    const afterMatchedTextIndex = match.index + matchedConditionalStartText.length
+
+    const matchingClosingBracketIndex = findMatchingClosingBracketIndex(source, afterMatchedTextIndex)
+    if (matchingClosingBracketIndex < 0) {
+      // TODO: warn user?
+      continue
+    }
+
+    let conditionalEndingIndex = matchingClosingBracketIndex
+
+    const trueBlockContent = source.substring(afterMatchedTextIndex, matchingClosingBracketIndex)
+    let falseBlockContent = ""
+
+    conditionalElseStartRegex.lastIndex = matchingClosingBracketIndex
+    const elseMatch = conditionalElseStartRegex.exec(source)
+    if (elseMatch != null) {
+      const afterElseMatchedTextIndex = elseMatch.index + elseMatch[0].length
+      const matchingClosingBracketForElseIndex = findMatchingClosingBracketIndex(source, afterElseMatchedTextIndex)
+      if (matchingClosingBracketForElseIndex > 0) {
+        falseBlockContent = source.substring(afterElseMatchedTextIndex, matchingClosingBracketForElseIndex)
+        conditionalEndingIndex = matchingClosingBracketForElseIndex
+      } // TODO: else warn user?
+    }
+
+    let passed = null
+    const sanitizedLeft = left.trimEnd()
+    const sanitizedRight = right.trimStart()
+    switch (operation) {
+      case "==": {
+        passed = sanitizedLeft === sanitizedRight
+        break
+      }
+      case "!=": {
+        passed = sanitizedLeft !== sanitizedRight
```

Comment:
Let's stick to `!=` instead of `!==` since we don't use it anywhere else. In this case it will only ever be strings any way.